hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/impl/ReplicaMap.java

```diff
@@ -184,7 +203,7 @@ void mergeAll(ReplicaMap other) {
   ReplicaInfo remove(String bpid, Block block) {
     checkBlockPool(bpid);
     checkBlock(block);
-    try (AutoCloseableLock l = writeLock.acquire()) {
+    try (AutoCloseDataSetLock l = lockManager.readLock(LockLevel.BLOCK_POOl, bpid)) {
```

Comment:
I agree with @Hexiaoqiao . We should keep the read/write lock. Once `block Pool lock` is introduced, we can discuss whether `write locks` need to be changed to `read locks` in other JIRA.

Reply:
Now the LightWeightResizableGSet is thread-safe, so most operate in ReplicaMap just need block pool read.